age/keysource.go

```diff
@@ -222,11 +228,109 @@ func (key *MasterKey) ToMap() map[string]interface{} {
 	return out
 }
 
+// Tries to find the public key file given the path of the private ssh key file
+func readPublicKeyFile(privateKeyPath string) (ssh.PublicKey, error) {
+	publicKeyPath := privateKeyPath + ".pub"
+	f, err := os.Open(publicKeyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to obtain public %q key for %q SSH key: %w", publicKeyPath, privateKeyPath, err)
+	}
+	defer f.Close()
+	contents, err := ioutil.ReadAll(f)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read %q: %w", publicKeyPath, err)
+	}
+	pubKey, _, _, _, err := ssh.ParseAuthorizedKey(contents)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse %q: %w", publicKeyPath, err)
+	}
+	return pubKey, nil
+}
+
+// Parses a private ssh key into an age identity
+// If the key is password secured it needs to add a passphrase promt
+func getAgeSshIdentityFromPrivateKeyFile(keyPath string) (age.Identity, error) {
```

Comment:
```suggestion
func parseSSHIdentityFromPrivateKeyFile(keyPath string) (age.Identity, error) {
```

would match better with `parseRecipient`, etc. As the context of the key service is already "age".

Reply:
In addition, something like:

```
// parseSSHIdentityFromPrivateKeyFile returns an age.Identity from the given
// private key file. If the private key file is encrypted, it will configure
// the identity to prompt for a passphrase.